sambacc/ctdb.py

```diff
@@ -200,6 +203,19 @@ def ensure_ctdb_node_present(
     ensure_ctdb_nodes(nodes, real_path=real_path, canon_path=canon_path)
 
 
+def ensure_ctdb_nodes_script(iconfig: config.InstanceConfig) -> None:
+    ctdb_params = iconfig.ctdb_config()
+    if not ctdb_params.get("nodes_cmd"):
+        return
+    cmd = ctdb_params["nodes_cmd"]
+    with open(cmd, "w") as fh:
+        fh.write("#!/bin/sh\n")
+        fh.write("exec samba-container ctdb-list-nodes\n")
+        fh.flush()
+        os.fchmod(fh.fileno(), 0o755)
+        os.fsync(fh)
```

Comment:
Why are we creating another executable with the required command? Why don't we directly invoke `ctdb-list-nodes` command to _samba-container_? Or is this done to not to interfere with current execution environment's `STDOUT`?

Reply:
Right now, the patches to CTDB only support a single command name, no arguments. It was simpler for me to work around that on the sambacc side for now than it was to figure out the right way to do it in C. Help appreciated on that side: https://gitlab.com/phlogistonjohn/samba/-/commit/3d6efac6e8ae1056149fb14ada40f6ff5db0f7c6#61bd70ccace1b02c4b6dfa6bd251aadbd56de122_167_188

Reply:
Resolved in samba repo.